__tests__/app.test.js

```diff
@@ -14,100 +14,51 @@ afterAll(() => {
-      .get('/api/topics')
-      .expect(200)
-      .then(({ body: { topics } }) => {
-        expect(topics).toHaveLength(3);
-        topics.forEach((topic) => {
-          expect(topic).toHaveProperty("description");
-          expect(topic).toHaveProperty("slug");
-        });
-      });
-   }); 
-});
+describe('Testing for News app', () => {
 
-   describe("Sad paths :( ", () => {
-    it("404 for invalid paths", () => {
-      return request(app)
-        .get("/api/topicss")
-        .expect(404)
-        .then(({ body: { msg } }) => {
-          expect(msg).toBe('Page not found');
-        });
-    });
-  });
+  describe("5. PATCH /api/articles/:article_id", () => {
+    it("check the vote by passed newVote", () => {
+      const incrementVotes = { inc_votes: 1 };
 
-  describe("4. GET /api/articles/:article_id", () => {
-    it("check the properties of an article object", () => {
-      let ARTICLE_ID = 1;
       return request(app)
-        .get(`/api/articles/${ARTICLE_ID}`)
+        .patch('/api/articles/1')
+        .send(incrementVotes)
         .expect(200)
-        .then(({body:{ article }}) => {
-          expect(article).toHaveProperty("author");
-          expect(article).toHaveProperty("title");
-          expect(article).toHaveProperty("body");
-          expect(article).toHaveProperty("topic");
-          expect(article).toHaveProperty("created_at");
-          expect(article).toHaveProperty("votes");
-          expect(article).toHaveProperty("article_id");
-          expect(article.article_id).toEqual(1);
-        });
-    });
-
-    it("400: bad request response for invalid path", () => {
-        return request(app)
-        .get('/api/articles/notAnID')
-        .expect(400)
         .then(({ body }) => {
-            expect(body.msg).toBe("Invalid input");
+          expect(body.article).toEqual({
+            title: "Living in the shadow of a great man",
+            topic: "mitch",
+            author: "butter_bridge",
+            body: "I find this existence challenging",
+            created_at: "2020-07-09T20:11:00.000Z",
+            votes: 101,
+            article_id: 1,
+          });
         });
     });
+    it("400: bad request response for invalid path", () => {
+      const incrementVotes = { inc_votes: 99999999 };
 
-    it("404: bad request response for the vaild ID but the article does not exist in the database", () => {
       return request(app)
-      .get('/api/articles/99999')
-      .expect(404)
+      .patch('/api/articles/NotID')
+      .send(incrementVotes)
+      .expect(400)
       .then(({ body }) => {
-          expect(body.msg).toBe('Page not found');
+          expect(body.msg).toBe("Invalid input");
       });
   });
-  });
 
-//   Responds with:
-// - an array of objects, each object should have the following property: `username`,`name`, `avatar_url`
+  it("400: bad request for patch body without inc_votes", () => {
+    const incrementVotes = { inc_votes: "null" };
```

Comment:
Sorry Jeeann, I mean if the `inc_votes` property itself is not there. For example, if we say `const incrementVotes = {}`, this should be a 400, because `inc_votes` does not exist in the object. I hope that's clearer now!

Reply:
Added `err.code === "23502" ` and now it works! Thank you for your help :)